src/tests/Directory.Build.targets

```diff
@@ -448,4 +448,40 @@
   <!-- RunOnly projects have a special build for dependent projects -->
   <Import Project="$(MSBuildThisFileDirectory)Common\runonly.targets" Condition="'$(CLRTestKind)' == 'RunOnly'" />
 
+  <PropertyGroup Condition="'$(TestBuildMode)' == 'nativeaot'">
+    <!-- NativeAOT compiled output is placed into a 'native' subdirectory: we need to tweak
+         rpath so that the test can load its native library dependencies if there's any -->
+    <IlcRPath Condition="'$(TargetOS)' != 'OSX'">$ORIGIN/..</IlcRPath>
```

Comment:
Yes, I tried to capture it in the comment above this line - the test build places the NativeAOT compiled artifacts in a `native` subdirectory, but we might have .so/.dll files that the test depends on one level up. We tweak rpath so that it loads from there. On Windows, we search the current directory, so it seems to work fine.